examples/source/style-layout/Dark_Mode_Toggle.html

```diff
@@ -0,0 +1,107 @@
+  as well as a manual toggle for people on non-supporting browsers.
+  You can read more about `prefers-color-scheme` in the article
+  https://web.de/prefers-color-scheme.
+-->
+
+<!-- -->
+<!doctype html>
+<html ⚡>
+  <head>
+    <meta charset="utf-8">
+    <link rel="canonical" href="./Dark_Mode_Toggle.html">
+    <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+    <script async src="https://cdn.ampproject.org/v0.js"></script>
+    <title>Dark Mode Sample</title>
+    <style amp-custom>
+      /**
+       * We need a global wrapper since we work with the adjacent selector below.
+       * The wrapper needs to cover the entire viewport.
+       */
+      .wrapper {
+        --padding: 0.25rem;
+
+        color: black;
+        background-color: white;
+        padding: var(--padding);
+        min-height: calc(100vh - 2 * var(--padding));
+      }
+
+      /**
+       * If the user states they prefer dark, obey.
+       * Don't offer the option to override.
+       */
+      @media (prefers-color-scheme: dark) {
+        .wrapper {
+          color: white;
+          background-color: black;
+        }
+
+        #dark-mode-checkbox,
+        #dark-mode-label {
+          display: none;
+        }
+      }
+
+      /**
+       * If the checkbox is checked, all adjacent siblings will be in dark mode.
+       * This will only be possible if the user doesn't prefer dark,
+       * or on browsers that don't understand `prefers-color-scheme`.
+       */
+      #dark-mode-checkbox:checked ~ * {
+        color: white;
+        background-color: black;
+      }
+
+      /**
+       * Position these absolutely so they don't "consume" space.
+       */
+      #dark-mode-checkbox,
+      #dark-mode-label {
+        position: absolute;
+      }
+
+      /**
+       * Don't display the ugly checkbox. On a real site, you could style the label
+       * as a button and work with the `:checked ::before` selector of the checkbox
+       * to convey the state.
+       */
+      #dark-mode-checkbox {
+        display: none;
+      }
+    </style>
+    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+  </head>
+  <body>
+    <!-- Everything _adjacent_ to this checkbox can be styled, leave it here in the DOM tree. -->
```

Comment:
one tip: wrap all `input`, `label` and `wrapper` in a single div. This way the whole code will be displayed in the doc view of the sample.

Reply:
Are you sure this works? In my tests, when the viewport is larger than the content wrapper, the underlying body background bleeds through.

![Screen Shot 2019-12-13 at 13 38 18](https://user-images.githubusercontent.com/145676/70800955-d945cf00-1dad-11ea-921a-b0eb0b05521d.png)

I load the page alone, not in the context of amp.dev.

Reply:
you're right, it works in amp.dev, but not standalone, but otherwise the inline preview would be broken.

<img width="880" alt="Screen Shot 2019-12-13 at 13 40 46" src="https://user-images.githubusercontent.com/380472/70801085-30e43a80-1dae-11ea-9d46-30078917d25a.png">

Reply:
Alright, thanks for confirming.